daemon/start.go

```diff
@@ -153,7 +153,15 @@ func (daemon *Daemon) containerStart(container *container.Container, checkpoint
 
 	spec, err := daemon.createSpec(container)
 	if err != nil {
-		return errdefs.System(err)
+		// daemon.createSpec has been updated to return all errdefs errors.
+		// However, this update may have been incomplete, and some things may
+		// have changed. To ensure that, no matter what, we get a valid error
+		// type, we will match on all known error types, and then wrap this
+		// error in "Unknown" if it is none of them
+		if !errdefs.IsKnownErrorType(err) {
```

Comment:
Wondering if we need this, as we already log errors that don't have a type assigned, and treat them as a `500`; https://github.com/moby/moby/blob/72b0b0387d777328882b02224bb78ad59c0837e3/api/server/httputils/errors.go#L62-L65

(basically; if we didn't handle the error, it's a server error, so a 500)